src/welcome/index.ts

```diff
@@ -10,7 +10,18 @@ import { isWalkthroughEnabled } from "../utils/walkthrough";
 
 let welcomeView: vscode.WebviewPanel | undefined;
 
-export async function showWelcomeWebview(context: vscode.ExtensionContext, _operationId?: string, options?: any) {
+// from command palette
+export async function showWelcomeWebviewBeside(context: vscode.ExtensionContext, _operationId?: string, options?: {
+    firstTimeRun?: boolean;
+}) {
+    const newOptions = { openBeside: true, firstTimeRun: options?.firstTimeRun };
```

Comment:
Directly use viewColumn as the option?

`{ viewColumn: vscode.ViewColumn.Beside, firstTimeRun: options?.firstTimeRun }`

Reply:
> Directly use viewColumn as the option?

That's would be unstraightforward if triggered from url... E.g. previous `command://java.welcome?openBiside=true` would become `command://java.welcome?viewColumn=-2`. (because current value of the enum `ViewColumn.Beside` is `-2`.

Reply:
Hmm, honestly I didn't see any problem with `java.welcome?viewColumn=-2`. Maybe in the future we want to have a new different position? Or VS Code proposed a new position called `Bottom`, who knows.

Anyway, not a big problem, leave the choice to you.

Reply:
> in the future we want to have a new different position? Or VS Code proposed a new position called Bottom

I don't think vscode teams will easily change value of enum (though there's possibility). My point is, for future maintainers `openBeside=true` is more readable comparing with `viewColumn=-2`.

Reply:
Not change but add. Like, a new value `viewColumn.Bottom` and user request that